lua/r/debug.lua

```diff
@@ -0,0 +1,199 @@
+------------------------------------------------------------------------------
+-- Support for debugging R code
+------------------------------------------------------------------------------
+local config = require("r.config").get_config()
+
+local M = {}
+
+-- Handle ambiwidth option and define signs
+if vim.o.ambiwidth == "double" then
+    vim.fn.sign_define(
+        "dbgline",
+        { text = "==>", texthl = "SignColumn", linehl = "QuickFixLine" }
+    )
+else
+    vim.fn.sign_define(
+        "dbgline",
+        { text = "▬▶", texthl = "SignColumn", linehl = "QuickFixLine" }
+    )
+end
+
+local s = {
+    debugging = false,
+    bufnr = -1,
+    winnr = -1,
+    func_offset = -2, -- Did not seek yet
+    fnm = "",
+}
+
+local switch_buf = function(fnm)
+    if vim.api.nvim_get_current_buf() ~= require("r.edit").get_rscript_buf() then
+        vim.cmd("sb " .. tostring(require("r.edit").get_rscript_buf()))
+    end
+    if vim.bo.modified then vim.cmd("split") end
+    vim.cmd("edit " .. fnm)
+    s.bufnr = vim.api.nvim_get_current_buf()
+    s.winnr = vim.api.nvim_get_current_win()
+end
+
+local find_func = function(srcref)
+    local rlines
+    s.func_offset = -1 -- Not found
+
+    vim.wait(300)
+    if type(config.external_term) == "boolean" and not config.external_term then
+        rlines = vim.api.nvim_buf_get_lines(require("r.term").get_buf_nr(), 0, -1, false)
+    else
+        local run_cmd = {
+            "tmux",
+            "-L",
+            "Rnvim",
+            "capture-pane",
+            "-p",
+            "-t",
+            require("r.external_term").get_tmuxsname(),
+        }
+        local resp = require("r.utils").system(run_cmd, { text = true }):wait()
```

Comment:
This is the faulty line when trying to source a file that includes a `browser()` call.